nixos/modules/services/backup/restic.nix

```diff
@@ -298,7 +298,9 @@ in
             doBackup = (backup.dynamicFilesFrom != null) || (backup.paths != null && backup.paths != []);
             pruneCmd = optionals (builtins.length backup.pruneOpts > 0) [
               (resticCmd + " forget --prune " + (concatStringsSep " " backup.pruneOpts))
-              (resticCmd + " check " + (concatStringsSep " " backup.checkOpts))
+            ];
+            checkCmd = optionals (builtins.length backup.checkOpts > 0) [
+                (resticCmd + " check " + (concatStringsSep " " backup.checkOpts))
             ];
```

Comment:
This check is annoying, as it is valid to call `restic check` without check specific options.
You should introduce a new Option to configure whether `check` is run. (For ease of use the default value of that option can then depend on `builtins.length backup.checkOpts > 0`)

Reply:
:man_facepalming: I completely forgot the 'default' check version. I have added a `runCheck` option that defaults to `length backups.checkOpts > 0`.